### PR TITLE
🚀 [feat] : setting up routes for authentication

### PR DESCRIPTION
**Pull request by Bismay5467**
DONE : 

- setup a basic page for registration, sign in, sign out
- setup routes to handle registration of new user, login and logout
- JWT based authentication (stored in cookies)
- forget password page (The password can be reset by clicking on the link sent in the email)

TODO : 

- OAuth using next-auth (a NPM lib) [though I need to figure out how to handle things in production. It seems google has some strict verification rules for external OAuth]

@Shubham-Rasal I am facing some issue in validating password in the sign in route. I need to have a look at the issue. I would suggest not to merge this PR right now. 

@VaibhavNITK you may start styling these pages by pulling this branch in your machine.

@Shubham-Rasal @VaibhavNITK feel free to review this PR and post your thoughts in the comment section.

### CODE REVIEW

File: exam-archive/.env.development.local
Comment (Shubham-Rasal) [inline exam-archive/.env.development.local:1]: Never commit env files to a PR. Try to make it a habit to ignore them using .gitignore

**Comment by Bismay5467** [comment]
image

@Shubham-Rasal

**Comment by Shubham-Rasal** [comment]
It is a problem only with OAuth or even with the plain old JWT-based auth as well? Also error logs with possible places the error  might be occurring can be really useful.

File: exam-archive/.env.development.local
Comment (Bismay5467) [reply exam-archive/.env.development.local:1]: Yeah, ik. I committed that the .env file so that you guys can test the API routes.

**Comment by Bismay5467** [comment]
> It is a problem only with OAuth or even with the plain old JWT-based auth as well? Also error logs with possible places the error might be occurring can be really useful.

This is a problem that I am facing in the database section. It has nothing to do with the authentication part. You may make a API call to the sign in route (POST) with the apt request body to reproduce the error.

**Comment by Bismay5467** [comment]
@Shubham-Rasal 

The API request will look something similar to this

curl --location 'http://localhost:3000/api/auth/signIn' \
--header 'Content-Type: application/json' \
--data-raw '{
    
    "username": "abc@gmail.com",
    "password": "qwer1@"
}'

**Comment by Bismay5467** [comment]
@Shubham-Rasal I am yet to commit the reset password page. Not yet figured out how to fix the rendering issue. Right now, I am working on it. Beside that, other features seem to working as expected. Can you please confirm once ? Maybe I may missed some edge cases? Or any vulnerability?

**Comment by Bismay5467** [comment]
@Shubham-Rasal I've come up with a workaround for the rendering problem, however I'm sure it's not the ideal solution. Let's see if @VaibhavNITK can offer any advice on how to handle it. The password reset page has been committed by me. 

@Shubham-Rasal @VaibhavNITK I would advise you everyone to test each of the parts and APIs in this PR and let me know what you think. Any and all comments would be much valued.  

@VaibhavNITK There is a TODO about creating a 403 error page in the reset page (under the auth folder).

@Shubham-Rasal  I think I've finished my part. If there are no modifications required, you can merge this PR right away. In a later PR, I would use the OAuth feat. Let's just stay with the current JWT based authentication

**Pull request merged**